docs/wallet-integration-guide/src/preparing-and-signing-transactions/index.rst

```diff
@@ -51,30 +52,34 @@ the underlying code that creates the command is:
             ]
         }
 
-preparing the Transaction
+Preparing the Transaction
 -------------------------
-Now that we have a command we need to prepare the transaction against the ledger. A transaction is a collection of commands that are atomic, meaning that either all commands
+Now that we have a command we need to prepare the transaction against the ledger which will return an unsigned transaction. A transaction is a collection of commands that are atomic, meaning that either all commands
 succeed or none of them do. To prepare a transaction we need to send the commands to the ledger:
 
 .. literalinclude:: ../../examples/snippets/prepare-ping-transaction.ts
     :language: typescript
     :dedent:
 
-The return type is a transaction if the combination of the commands are possible, otherwise an error is returned.
+The return type is an unsigned transaction if the combination of the commands are possible, otherwise an error is returned. The transaction can then be visualised and signed by the party.
 
 Validating the Transaction
 --------------------------
 The transaction is returned alongside with the hash that needed to be signed. If the validator is not controlled by you, then it might
-be a good idea to validate that the transaction is what you expect it to be. You can use the Wallet SDK to visualize the transaction (as described in the ../finding-and-reading-data/index.rst#visualizing-a-transaction section).
-On top of visualizing the transaction confirming the hash matches before signing is also valuable.
+.. What exactly is returned here? A transaction and a hash? How are they returned? In the same JSON response?
+be a good idea to validate that the transaction is what you expect it to be. You can use the Wallet SDK to visualize the transaction 
+as described in the :ref:`Visualizing a transaction section <visualizing-a-transaction>`.
+
+On top of visualizing the transaction, it's also important to compute the transaction hash yourself and confirm that it matches the hash of the transaction provided by the validator from the prepare step.
 
 The hash can be computed using the Wallet SDK:
 
 .. literalinclude:: ../../examples/snippets/compute-transaction-hash.ts
     :language: typescript
     :dedent:
 
-you can then compare the `hash` with the `transaction.preparedTransactionHash` to ensure they match.
+You can then compare the `hash` with the `transaction.preparedTransactionHash` to ensure they match.
+.. Can we provide an example of this? Do we have a link to the API?
```

Comment:
@PHOL-DA this is a question for you.

Reply:
the file snippet above has an example using the SDK